packages/openapi-generator/src/parser/unique-naming.ts

```diff
@@ -1,24 +1,75 @@
-import { UniqueNameGenerator, camelCase } from '@sap-cloud-sdk/util';
+import { UniqueNameGenerator, camelCase, unique } from '@sap-cloud-sdk/util';
+import { ParserOptions } from './options';
 
 /**
- * Ensure uniqueness of names.
- * Takes a list of names and renames those that are duplicate.
- * The renamed list has the same order as the original list.
- * @param names List of names to deduplicate.
- * @param options Object containing options to configure the transformation.
+ * Format names and ensure uniqueness of names.
+ * With `strictNaming` enabled, duplicate names lead to an error.
+ * With `strictNaming` disabled, duplicate names are renamed.
+ * The formatted (and renamed) list of names has the same order and length as the original list.
+ * @param names List of names to ensure uniqueness for.
+ * @param options Parser options.
+ * @param namingOptions Object containing options to configure the transformation.
  * @param options.format Function to format the name. Defaults to camel case.
  * @param options.reservedWords Reserved words that should be handled as duplicates.
  * @returns Unique names in the given order.
  */
 export function ensureUniqueNames(
   names: string[],
-  options: {
+  options: ParserOptions,
+  namingOptions: {
```

Comment:
Since you created `ParserOptions`, should we also have a `NamingOptions`?

Reply:
Yes, I thought about this, but it is only used in two places... what do you think?

Reply:
This is a minor comment, you decide;)

Reply:
Then I would leave it unless it becomes problematic or @FrankEssenberger also wants this ;)